src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs

```diff
@@ -143,14 +190,181 @@ public class DependencyGraph
+                        b => b.Channels != null && b.Channels.Any());
+                    // If no build was found (e.g. build was flowed without a channel or channel was removed from
+                    // a build, then no diff from latest.
+                    if (newestBuildWithChannel != null)
+                    {
+                        int channelId = newestBuildWithChannel.Channels.First().Id.Value;
+                        // Just choose the first channel. This algorithm is mostly just heuristic.
+                        string latestCommitKey = $"{node.RepoUri}@{channelId}";
+                        string latestCommit = null;
+                        if (!latestCommitCache.TryGetValue(latestCommitKey, out latestCommit))
+                        {
+                            // Look up latest build in the channel
+                            var latestBuild = await barOnlyRemote.GetLatestBuildAsync(node.RepoUri, channelId);
+                            // Could be null, if the only build was removed from the channel
+                            if (latestBuild != null)
+                            {
+                                latestCommit = latestBuild.Commit;
+                            }
+                            // Add to cache
+                            latestCommitCache.Add(latestCommitKey, latestCommit);
+                        }
+
+                        // Perform diff if there is a latest commit.
+                        if (!string.IsNullOrEmpty(latestCommit))
+                        {
+                            IRemote repoRemote = remoteFactory.GetRemote(node.RepoUri, logger);
+                            // This will return a no-diff if latestCommit == node.Commit
+                            node.DiffFrom = await repoRemote.GitDiffAsync(node.RepoUri, latestCommit, node.Commit);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Diff each node in the graph against the latest build in
+        ///     the graph.
+        /// </summary>
+        /// <param name="remoteFactory"></param>
+        /// <param name="logger"></param>
+        /// <param name="nodeCache"></param>
+        /// <param name="visitedRepoUriNodes"></param>
+        /// <returns></returns>
+        private static async Task DoLatestInGraphNodeDiffAsync(
+            IRemoteFactory remoteFactory,
+            ILogger logger,
+            Dictionary<string, DependencyGraphNode> nodeCache,
+            Dictionary<string, DependencyGraphNode> visitedRepoUriNodes)
+        {
+            logger.LogInformation("Running latest in graph node diff.");
+
+            // Find the build of each repo in the graph, then
+            // get the diff info from the latest
+            Dictionary<string, DependencyGraphNode>.KeyCollection uniqueRepos = visitedRepoUriNodes.Keys;
```

Comment:
>Dictionary<string, DependencyGraphNode>.KeyCollection [](start = 12, length = 53)

I'd just call this one IEnumerable<string>. Or not even put it in a variable, since we only use it on the next line.